File: 3_Math/566_Reshape_the_Matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        int m = nums.size();
        int n = nums[0].size();
        if(r*c != m*n) return nums;
        vector<vector<int>> ans(r, vector<int>(c, 0));
        for(int i=0; i<m*n; ++i){
            int org_x = i/n;
            int org_y = i%n;
            int new_x = i/c;
            int new_y = i%c;
            ans[new_x][new_y] = nums[org_x][org_y];
        }
        return ans;
    }
};

File: S_Math/Geometry/566_Reshape_the_Matrix.py
""" L1
From range m*n, indexes are only determined by column number.
"""


class Solution:
    def matrixReshape(self,
                      nums: List[List[int]],
                      r: int,
                      c: int) -> List[List[int]]:
        if r * c > len(nums) * len(nums[0]):
            return nums
        m, n = len(nums), len(nums[0])
        ans = [[0 for _ in range(c)] for _ in range(r)]
        for i in range(m * n):
            ans[int(i / c)][int(i % c)] = nums[int(i / n)][int(i % n)]
        return ans
